src/features/DisplayControls/PatientHeader/tests/PatientHeader.spec.jsx

```diff
@@ -1,52 +1,236 @@
+          "display": "Test Father"
+        }
+      },
+      {
+        "uuid": "48aa82ca-3711-40ed-b0f1-bc05013ea803",
+        "display": "Samuel is the Mother of Mother",
+        "relationshipType": {
+          "uuid": "c86d9979-b8ac-4d8c-85cf-cc04e7f16002",
+          "display": "Mother/Patient"
+        },
+        "personB": {
+          "uuid": "c732b222-6b9c-494b-ab3d-faea648e004c",
+          "display": "Mother Test"
+        }
+      }
+    ]
+    })
+   mockFetchAddressMapping.mockResolvedValue(
+    [
+      {
+          "name": "Country",
+          "addressField": "country",
+          "required": false
+      },
+      {
+          "name": "Region",
+          "addressField": "address1",
+          "required": false
+      },
+      {
+          "name": "Zone",
+          "addressField": "address2",
+          "required": false
+      },
+      {
+          "name": "Woreda",
+          "addressField": "address3",
+          "required": false
+      },
+      {
+          "name": "Kebele",
+          "addressField": "address4",
+          "required": false
+      },
+      {
+          "name": "Address",
+          "addressField": "address5",
+          "required": false
+      }
+  ]) 
   });
 
   afterEach(() => {
     jest.resetAllMocks();
   });
 
-  it("renders without crashing", () => {
+  it("should render without crashing", () => {
     render(<PatientHeader patientId="123" />);
   });
 
-  it("calls fetchPatientInfo on mount", () => {
+  it("should call fetchPatientInfo on mount", () => {
     render(<PatientHeader patientId="123" />);
-    expect(fetchPatientInfo).toHaveBeenCalledWith("123");
+    expect(mockFetchPatientProfile).toHaveBeenCalledWith("123");
   });
 
-  it("displays loading skeleton while fetching data", () => {
+  it("should display loading skeleton while fetching data", () => {
     render(<PatientHeader patientId="123" />);
     expect(screen.getByTestId("header-loading")).toBeTruthy();
   });
 
-  it("displays patient details after data is fetched", async () => {
+  it("should display patient details after data is fetched", async () => {
     render(<PatientHeader patientId="123" />);
     await waitFor(() => expect(screen.getByText("John Doe")).toBeTruthy());
     expect(screen.getByText(/30 Years/i)).toBeTruthy();
     expect(screen.getByText("01/01/1991")).toBeTruthy();
     expect(screen.getByText(/12345/i)).toBeTruthy();
   });
+
+  it("should display all details of the patient", async () => {
+     const {container} = render(<PatientHeader patientId="123"/>);
+    await waitFor(() => 
+    expect(screen.getByText("John Doe")).toBeTruthy());
+    const showDetailsButton = screen.getByText("Show Details");
+    fireEvent.click(showDetailsButton);
+
+    await waitFor (() =>
+    {
+      expect(screen.getByText(/Country : Ethiopia/i)).toBeTruthy();
+      expect(container).toMatchSnapshot();
+    })
+    // expect(screen.getByText(/Residence Phone : 22222222/i)).toBeTruthy();
```

Comment:
can we have these asserts as well? so that we can be sure only correct data is displayed in the ui

Reply:
snapshot is added for the same . do we need it then?